lib/rules/no-else-return.js

```diff
@@ -12,6 +12,20 @@
 const astUtils = require("../ast-utils");
 const FixTracker = require("../util/fix-tracker");
 
+/**
+ * Get allowElseIf config value safely.
+ *
+ * @param {Object} context The eslint context
+ * @returns {boolean} True if allowElseIf or anything on path undefined
+ */
+function getAllowElseIf(context) {
+    try {
+        return context.options[0].allowElseIf;
```

Comment:
This will work incorrectly if an empty object is provided as an option:

```js
/* eslint no-else-return: [error, {}] */
```

That config should result in the default `allowElseIf` value getting used (i.e. `allowElseIf: true`). However, with the current implementation, that config will cause this function to return `undefined`, which will get treated as `false`.

Additionally, using `try`/`catch` to check for property existence seems like an antipattern to me, because it will end up also swallowing unrelated errors, and a missing option is not an "exceptional" case.

I would recommend using something like the code snippet from https://github.com/eslint/eslint/pull/9229#discussion_r141237682 to check the `allowElseIf` option.

Reply:
@not-an-aardvark but... but.... lodash 😢. Made the change.

Reply:
Thanks, I appreciate it.

For what it's worth, there are a couple reasons I didn't want the lodash solution to be used:

* The main reason is that it would require contributors to have read the lodash documentation in order to understand that code.
* As a personal preference, lodash's strategy of accepting `"context.options[0]"` as a *string* seems messy to me -- it feels like it shouldn't be necessary for a library to do parsing on property accessors just to access a property safely.

Doing something like `foo && foo.bar && foo.bar.baz` is admittedly a bit messy, but it's idiomatic JavaScript and I think it's the best solution at the moment. (Hopefully in the future we'll be able to use [optional chaining](https://github.com/tc39/proposal-optional-chaining) to make it less messy.)